lib/logstash/filters/truncate.rb

```diff
@@ -79,18 +83,23 @@ def trim(value, length)
       return v
     end
 
-    def truncate_all(event, length)
-      # TODO(sissel): I couldn't find a better way to get the top level keys for
-      # an event. Nor could I find a way to iterate over all the keys in an
-      # event, so this may have to suffice.
-      fields = event.to_hash.keys.map { |k| "[#{k}]" }
+    def truncate_all(fields, event, length)
+      truncated = []
```

Comment:
Instead of using an Array to check if any value is `true` I think we could use an accumulator of the partial result using the `or` operator.
```suggestion
      truncated = false
```